CRM/Event/Form/ManageEvent/Location.php

```diff
@@ -249,18 +249,21 @@ public function postProcess() {
           $params[$block][$index]['id'] = $this->locationBlock['loc_block.' . $block . $fieldKey];
         }
       }
-      if ($block !== 'address') {
-        $location[$block] = CRM_Core_BAO_Block::create($block, $params, 'event');
-      }
-      else {
-        $location[$block] = CRM_Core_BAO_Address::create($params, TRUE, 'event');
-      }
     }
-
-    $params['loc_block_id'] = CRM_Core_BAO_Location::createLocBlock($location, [
-      'entity_table' => $params['entity_table'],
-      'entity_id' => $params['entity_id'],
-    ]);
+    $addresses = Address::save(FALSE)->setRecords($params['address'])->execute();
+    $emails = Email::save(FALSE)->setRecords($params['email'])->execute();
+    $phones = Phone::save(FALSE)->setRecords($params['phone'])->execute();
```

Comment:
Just sayin' - this APIv4 code looks really slick. It's an API but it also feels more "native" than constructing a DAO, calling `copyValues()` and `save()`. It's like it's what the DAO should have been.